Controller/Request/Post.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace SomethingDigital\AuctionItem\Controller\Request;
+
+use Composer\DependencyResolver\Request;
+use Magento\Backend\Model\View\Result\RedirectFactory;
+use Magento\Framework\App\Action\HttpPostActionInterface;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Message\ManagerInterface;
+use SomethingDigital\AuctionItem\Model\AuctionItemFactory;
+use SomethingDigital\AuctionItem\Model\AuctionItemRepository;
+use Magento\Framework\Stdlib\DateTime\DateTimeFactory;
+
+/**
+ * Custom Controller Auction Item Post
+ *
+ * Class Index
+ */
+class Post implements HttpPostActionInterface
+{
+
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
+     * @var RedirectFactory
+     */
+    protected $redirectFactory;
+
+    /**
+     * @var ManagerInterface
+     */
+    protected $messageManager;
+
+    /**
+     * @var AuctionItemRepository
+     */
+    protected $auctionItemRepository;
+
+    /**
+     * @var AuctionItemFactory
+     */
+    protected $auctionItemFactory;
+
+    /**
+     * @var DateTimeFactory
+     */
+    protected $dateTimeFactory;
+
+    /**
+     * @param RequestInterface $request
+     */
+    public function __construct(
+        RequestInterface $request,
+        RedirectFactory $redirectFactory,
+        ManagerInterface $messageManager,
+        AuctionItemRepository $auctionItemRepository,
+        AuctionItemFactory $auctionItemFactory,
+        DateTimeFactory $dateTimeFactory
+    ) {
+        $this->request = $request;
+        $this->redirectFactory = $redirectFactory;
+        $this->messageManager = $messageManager;
+        $this->auctionItemRepository = $auctionItemRepository;
+        $this->auctionItemFactory = $auctionItemFactory;
+        $this->dateTimeFactory = $dateTimeFactory;
+    }
+
+    /**
+     * @return ResultInterface
+     */
+    public function execute()
+    {
+        $post = $this->request->getParams();
+        if ($post) {
+            $this->handlePost($post);
+        }
+
+        $redirect = $this->redirectFactory->create();
+        $redirect->setUrl('/auction/request/form');
+
+        return $redirect;
+    }
+
+    /**
+     * @return Void
+     */
+    public function handlePost($post)
+    {
+        $auctionItem = $this->auctionItemFactory->create();
+
+        $auctionItem->setItemName($post["item-name"]);
+        $auctionItem->setItemSku($post["item-sku"]);
+        $auctionItem->setItemBaseAuctionPrice($post["item-base-auction-price"]);
+        $auctionItem->setItemOwnerEmail($post["item-owner-email"]);
```

Comment:
It's good that we have some client-side validation on these values. Since this is user input that we're inserting directly into the DB, we should add some server-side validation as well. You can use the `/Zend_Validate` classes to check that each of the fields actually has the values it should. 
Here's a usage example from core:
https://github.com/magento/magento2/blob/c998238c9e2b2818ec567d2ed7df67331e4f964c/app/code/Magento/Backend/Model/Menu/Item/Validator.php#L43
https://github.com/magento/magento2/blob/c998238c9e2b2818ec567d2ed7df67331e4f964c/app/code/Magento/Backend/Model/Menu/Item/Validator.php#L151